ext/mysqli/mysqli_api.c

```diff
@@ -92,9 +91,12 @@ mysqli_escape_string_for_tx_name_in_comment(const char * const name)
 				v == '=')
 			{
 				*p_copy++ = v;
-			} else if (warned == FALSE) {
-				php_error_docref(NULL, E_WARNING, "Transaction name truncated. Must be only [0-9A-Za-z\\-_=]+");
-				warned = TRUE;
+			} else {
+				zend_string *function_name = get_active_function_or_method_name();
+				zend_value_error("%s(): The transaction name can only contain the A-Z, a-z, 0-9, \"\\\", \"-\", \"_\", and \"=\" characters", ZSTR_VAL(function_name));
+				zend_string_release(function_name);
+				efree(p_copy);
+				return NULL;
```

Comment:
This is libmysql-only code. It either needs a corresponding change to mysqlnd code *somewhere*, or should not be changed.

Reply:
I swear I checked it.. :') But now I also see that it's libmysql-only indeed. That said, I'll revert throwing exception, but I think the error message change is worth to keep.

Reply:
Sorry, had to force push.